docs/source/tutorials/fine_tune_llama_7b.mdx

```diff
@@ -0,0 +1,356 @@
+
+```python
+from transformers import AutoTokenizer
+
+# Hugging Face model id
+model_id = "philschmid/Llama-2-7b-hf" # ungated
+# model_id = "meta-llama/Llama-2-7b-hf" # gated
+
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+```
+
+To pack/stack our dataset we need to first tokenize it and then we can pack it with the `pack_dataset` method. To prepare our dataset we will now:
+
+1. Format our samples using the template method and add an EOS token at the end of each sample
+2. Tokenize our dataset to convert it from text to tokens
+3. Pack our dataset to 2048 tokens
+
+```python
+from random import randint
+# add utils method to path for loading dataset
+import sys
+sys.path.append("./scripts/utils") # make sure you change this to the correct path
+from pack_dataset import pack_dataset
+
+
+# template dataset to add prompt to each sample
+def template_dataset(sample):
+    sample["text"] = f"{format_dolly(sample)}{tokenizer.eos_token}"
+    return sample
+
+# apply prompt template per sample
+dataset = dataset.map(template_dataset, remove_columns=list(dataset.features))
+# print random sample
+print(dataset[randint(0, len(dataset))]["text"])
+
+# tokenize dataset
+dataset = dataset.map(
+    lambda sample: tokenizer(sample["text"]), batched=True, remove_columns=list(dataset.features)
+)
+
+# chunk dataset
+lm_dataset = pack_dataset(dataset, chunk_length=2048) # We use 2048 as the maximum length for packing
+```
+
+After we processed the datasets we are going save it to disk. You could also save it to S3 or the Hugging Face Hub for later use.
+
+_Note: Packing and preprocessing your dataset can be run outside of the Trainium instance._
+
+```python
+# save train_dataset to disk
+dataset_path = "tokenized_dolly"
+lm_dataset.save_to_disk(dataset_path)
+```
+
+## 3. Fine-tune Llama on AWS Trainium using the `NeuronTrainer`
+
+Normally you would use the **[Trainer](https://huggingface.co/docs/transformers/v4.19.4/en/main_classes/trainer#transformers.Trainer)** and **[TrainingArguments](https://huggingface.co/docs/transformers/v4.19.4/en/main_classes/trainer#transformers.TrainingArguments)** to fine-tune PyTorch-based transformer models.
+
+But together with AWS, we have developed a `NeuronTrainer` to improve performance, robustness, and safety when training on Trainium instances. The `NeuronTrainer` is part of the `optimum-neuron` library and can be used as a 1-to-1 replacement for the `Trainer`.
+
+When it comes to distributed training on AWS Trainium there is a few things we need to take care of. Since Llama is a big model it might not fit on a single accelerator, thats why we added support for different distributed training strategies to the `NeuronTrainer` including:
```

Comment:
```suggestion
When it comes to distributed training on AWS Trainium there are a few things we need to take care of. Since Llama is a big model it might not fit on a single accelerator, thats why we added support for different distributed training strategies to the `NeuronTrainer` including:
```